### ISSUE TITLE
[BUG] make_reduction 'Level type mismatch' when fitting model

### ISSUE THREAD

**Issue by vattyamp** [alan-turing-institute/sktime#3344]
**Describe the bug**

I experienced this error when using make_reduction with any window size to fit and predict. When running `.predict`, it seems to be slicing indicies that are not in the test set. My date range for the test set goes from  2022-05-19 to 2022-05-25 but the error is with the date 2022-05-12, which is the the window_length number of days before the start day. The data is in hierarchal format

<br class="Apple-interchange-newline">

**To Reproduce**

```python

from xgboost import XGBRegressor

y_train, y_test, x_train, x_test= temporal_train_test_split(y_hat,x_new,test_size=7) # The week of 5/19 - 5/25 is the test

regressor = XGBRegressor(objective='reg:squarederror', random_state=42)
forecaster = make_reduction(regressor,window_length=7,strategy='recursive')

pipe = TransformedTargetForecaster(steps=[
    ("detrender", Deseasonalizer()),
    ("forecaster", forecaster)
])

pipe.fit(y=y_train,X=x_train)
y_pred = pipe.predict(fh=[1,2,3,4,5,6,7],X=x_test) # From 5-19 to 5-25

```

_Error message:_

**TypeError: Level type mismatch: 2022-05-12**

TypeError                                 Traceback (most recent call last)
<command-1972184593868422> in <module>
      1 #fh = ForecastingHorizon()
----> 2 y_pred = forecaster.predict(fh=[1,2,3,4,5,6,7],X=x_test)
      3 

/databricks/python/lib/python3.8/site-packages/sktime/forecasting/base/_base.py in predict(self, fh, X)
    323         # we call the ordinary _predict if no looping/vectorization needed
    324         if not self._is_vectorized:
--> 325             y_pred = self._predict(fh=fh, X=X_inner)
    326         else:
    327             # otherwise we call the vectorized version of predict

/databricks/python/lib/python3.8/site-packages/sktime/forecasting/base/_sktime.py in _predict(self, fh, X)
     29         # all values are out-of-sample
     30         if fh.is_all_out_of_sample(self.cutoff):
---> 31             return self._predict_fixed_cutoff(
     32                 fh.to_out_of_sample(self.cutoff), **kwargs
     33             )
/databricks/python/lib/python3.8/site-packages/sktime/forecasting/base/_sktime.py in _predict_fixed_cutoff(self, fh, X, return_pred_int, alpha)
     63         """
     64         # assert all(fh > 0)
---> 65         y_pred = self._predict_last_window(
     66             fh, X, return_pred_int=return_pred_int, alpha=alpha
     67         )
/databricks/python/lib/python3.8/site-packages/sktime/forecasting/compose/_reduce.py in _predict_last_window(self, fh, X, return_pred_int, alpha)
    665             y_last, X_last = self._get_shifted_window()
    666         else:
--> 667             y_last, X_last = self._get_last_window()
    668 
    669         # If we cannot generate a prediction from the available data, return nan.

/databricks/python/lib/python3.8/site-packages/sktime/forecasting/base/_sktime.py in _get_last_window(self)
    126 
    127         # Get the last window of the endogenous variable.
--> 128         y = self._y.loc[start:cutoff].to_numpy()
    129 
    130         # If X is given, also get the last window of the exogenous variables.

**Expected behavior**

Perform a global forecast on all the hierarchical data.

**Additional context**

I'm trying to transform only the `y` target variable and leave the `x` data alone. Not sure if I'm using make_reduction when I'm not supposed to be or if there's an equivalent

**Versions**
<details>

System:
    python: 3.8.10 (default, Jun 22 2022, 20:18:18)  [GCC 9.4.0]
executable: /local_disk0/.ephemeral_nfs/envs/pythonEnv-e16893be-13e7-4812-b22f-c0d78b237332/bin/python
   machine: Linux-5.4.0-1086-azure-x86_64-with-glibc2.29

Python dependencies:
          pip: 21.0.1
   setuptools: 52.0.0
      sklearn: 1.1.2
       sktime: 0.13.1
  statsmodels: 0.12.2
        numpy: 1.21.6
        scipy: 1.6.2
       pandas: 1.2.4
   matplotlib: 3.4.2
       joblib: 1.0.1
        numba: 0.55.1
     pmdarima: 1.8.4
      tsfresh: None

</details>

**Comment by fkiraly** [comment]
Thanks for reporting.
While we look into this, would you like to try out the new prototype that can to global fitting? https://github.com/alan-turing-institute/sktime/pull/3333

Feedback/ideas here: https://github.com/alan-turing-institute/sktime/issues/3224

**Comment by fkiraly** [comment]
Oh, and could you provide a self-contained example? I.e., including imports, creation of dummy data, etc, or it will be much harder to debug.

**Comment by vattyamp** [comment]
I created an example from scratch:
```
from sktime.datatypes import get_examples
import pandas as pd
import numpy as np
from sktime.forecasting.compose import make_reduction
from sktime.forecasting.compose import RecursiveTabularRegressionForecaster
from sktime.forecasting.model_selection import temporal_train_test_split
from xgboost import XGBRegressor

X = get_examples(mtype="pd_multiindex_hier")[0]

Y = X.drop('var_0',axis=1)

y_train, y_test, x_train, x_test= temporal_train_test_split(Y,X,test_size=1)

regressor = XGBRegressor(objective='reg:squarederror', random_state=42)
forecaster = make_reduction(regressor,window_length=1,strategy='recursive')

forecaster.fit(y=y_train,X=x_train)
y_pred = forecaster.predict(fh=1,X=x_test)
```
The data I used in this example was from the example set. The error I get when I run this block is

TypeError: Level type mismatch: 1

**Comment by fkiraly** [comment]
ah, this bug was fixed in 0.13.2.

Duplicate of #3316.

If you want a global fit, see #3333 (experimental).
